mongo/src/main/java/com/redhat/lightblue/mongo/crud/MongoSafeUpdateProtocol.java

```diff
@@ -233,23 +267,96 @@ public int getSize() {
                 findConcurrentModifications(results);
             }
         }
+        retry(results);
```

Comment:
Really minor, but I would call this method "retryIfNeeded" or "retryIfConcurrentUpdate"